# Repo: mada-bat-picornavirus for paper linked [here]()
Full and partial genomes described from mNGS data collected on subset of Malagasy bat fecal and urine samples
---
Folders and R scripts for final and supplemental figures are linked [here](https://github.com/brooklabteam/mada-bat-picornavirus/tree/main/final_figures)

Folders and R scripts for final and supplemental tables are linked [here](https://github.com/brooklabteam/mada-bat-picornavirus/tree/main/final_tables)
---

Fasta files, command line contig sorting, and reference sequences for phylogenies are [here](https://github.com/brooklabteam/mada-bat-picornavirus/tree/main/fasta_files)

Files for genome annotation and characterization are [here](https://github.com/brooklabteam/mada-bat-picornavirus/tree/main/genome_annotation_and_characterization)

Metadata files are [here](https://github.com/brooklabteam/mada-bat-picornavirus/tree/main/metadata)

ModelTest output is [here](https://github.com/brooklabteam/mada-bat-picornavirus/tree/main/modeltest)

RAxML trees are linked [here](https://github.com/brooklabteam/mada-bat-picornavirus/tree/main/raxml_trees)

Files and scripts for mapping reads is [here](https://github.com/brooklabteam/mada-bat-picornavirus/tree/main/read_mapping)

RDP4 recombination analysis files are [here](https://github.com/brooklabteam/mada-bat-picornavirus/tree/main/recombination)

Files to make a sampling map and summary files are [here](https://github.com/brooklabteam/mada-bat-picornavirus/tree/main/sampling_map)

PySimPlot files are [here](https://github.com/brooklabteam/mada-bat-picornavirus/tree/main/PySimPlot)

---
